samples/bluetooth/peripheral_cgm/src/main.c

```diff
@@ -0,0 +1,193 @@
+				BT_UUID_16_ENCODE(BT_UUID_DIS_VAL))
+};
+
+static void connected(struct bt_conn *conn, uint8_t err)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+	if (err) {
+		printk("Failed to connect to %s (err 0x%02x)\n", addr, err);
+	} else {
+		printk("Connected %s\n", addr);
+	}
+}
+
+static void disconnected(struct bt_conn *conn, uint8_t reason)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+	printk("Disconnected from %s (reason %u)\n", addr, reason);
+}
+
+static void security_changed(struct bt_conn *conn, bt_security_t level,
+				enum bt_security_err err)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+	if (!err) {
+		printk("Security changed: %s level %u\n", addr, level);
+	} else {
+		printk("Security failed: %s level %u err %d\n", addr, level,
+			err);
+	}
+}
+
+BT_CONN_CB_DEFINE(conn_callbacks) = {
+	.connected = connected,
+	.disconnected = disconnected,
+	.security_changed = security_changed,
+};
+
+static void auth_cancel(struct bt_conn *conn)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	printk("Pairing cancelled: %s\n", addr);
+}
+
+static void auth_passkey_display(struct bt_conn *conn, unsigned int passkey)
+{
+	printk("Pairing key is %06d.\n", passkey);
+}
+
+static struct bt_conn_auth_cb auth_cb_display = {
+	.cancel = auth_cancel,
+	.passkey_display = auth_passkey_display,
+};
+
+static void cgms_session_state_changed(const bool state)
+{
+	session_state = state;
+	if (state) {
+		printk("Session starts.\n");
+	} else {
+		printk("Session stops.\n");
+	}
+}
+
+void main(void)
+{
+	int err;
+	const uint8_t measurement_interval = 1; /* time in minutes. */
+	/* The time it will try to submit a record if last attempt fails. */
+	const uint8_t retry_count = 3;
+	/* The time interval between two attempts. Unit is second. */
+	const uint8_t retry_interval = 1;
+	float glucose = APP_GLUCOSE_MIN;
+	struct bt_cgms_measurement result;
+	struct bt_cgms_cb cb;
+	struct bt_cgms_init_param params;
+
+	printk("Starting Bluetooth Peripheral CGM example\n");
+
+	bt_conn_auth_cb_register(&auth_cb_display);
+
+	err = bt_enable(NULL);
+	if (err) {
+		printk("Bluetooth init failed (err %d)\n", err);
+		return;
+	}
+
+	printk("Bluetooth initialized\n");
+
+	if (IS_ENABLED(CONFIG_SETTINGS)) {
+		settings_load();
+	}
+
+	params.type = BT_CGMS_FEAT_TYPE_CAP_PLASMA;
+	params.sample_location = BT_CGMS_FEAT_LOC_FINGER;
+	/* The session will run 1 hour */
+	params.session_run_time = 1;
+	/* The initial communication interval will be 5 minutes */
+	params.initial_comm_interval = 5;
+	/* The shortest communication interval allowed client to change will be 1 minute.
+	 * The value should not be smaller than the time between two glucose measurements,
+	 * otherwise the adjacent two notifications may contain the same record.
```

Comment:
Is this not allowed by the service specification?
If not, `shortest_comm_interval` seems to be unnecessary. 
The record has a `time_offset`, so the user can distinguish the records.

Reply:
No explict expression. The spec says

> If the Operand that was sent with a Set CGM Communication Interval Op Code is 0x00, the Server may disable the periodic communication. If the Operand that was sent is 0xFF, the Server shall change its communication interval time to the fastest interval supported by the device. Otherwise the Server shall change its communication interval time according to the Operand value.

So I think there exists a "fastest interval supported by the device". If we need to define this value, it is reasonble to make sure it is not smaller than sampling interval.
Of course, as the app will take care of duplicate records, the sample will work with the app even if we remove this restriction.

Reply:
I am open to either way. See if you have any preference.

Reply:
It is easier to expand the API than to remove a item from it. So it seems to me that we can give up this parameter. If there will be a need in the future it can be added.

Reply:
OK. I will remove it. But we still need to define a minimum communication interval internally so that the operand of 0xFF can be handled, and in our case it should be 1 (minute). Should I make it configurable via Kconfig, or simply define a constant?

Reply:
I think it may be a constant.